Add Hall of Fame page to beta version

michaelrambeau opened the pull request:
## Goal

Add missing _Hall of Fame_ page from the current version: https://bestofjs.org/hall-of-fame

I was not sure about where to put the link to the Hall of Fame: I reproduced what we do in production, adding a "More" button in the site header, to reveal more links.

Instead of this button, we could use the "Navigation Menu" component: https://ui.shadcn.com/docs/components/navigation-menu

I like the fact that you don't have to click to reveal the navigation items with this "Navigation Menu", it's triggered by a mouse `hover`, it helps to discover all the links of the application.

I added a loading state but I realized we never see it when the application is built: the page has no search parameters, so it's statically built and I think the loading state never shows up.

## How to test

Click on "More" button in the header and enjoy the company of great people 😄 

## Screenshots

![image](https://github.com/bestofjs/bestofjs/assets/5546996/9c90ba6d-3274-4604-92d5-747cf2fe11af)


![image](https://github.com/bestofjs/bestofjs/assets/5546996/ef7e1e05-4681-4b13-8730-411b350456b5)

zawarudo commented on apps/bestofjs-nextjs/src/app/hall-of-fame/loading.tsx:
```diff
@@ -0,0 +1,39 @@
+import { Skeleton } from "@/components/ui/skeleton";
+import { PageHeading } from "@/components/core/typography";
+
+const memberNames = ["Evan", "Dan", "Sindre", "Ryan", "Kent", "TJ"];
```
Committed to the repo, what an honor!

zawarudo commented on apps/bestofjs-nextjs/src/lib/global.d.ts:
```diff
@@ -104,13 +104,19 @@ declare namespace BestOfJS {
 
-  type HallOfFameMember = {
-    username: string;
+  interface RawHallOfFameMember {
```
Raw means API's data model?

michaelrambeau replied:
> 

The `Raw*` objects are the ones we get from the JSON files (the "static API" that is rebuilt everyday)

zawarudo commented on apps/bestofjs-nextjs/src/lib/global.d.ts:
```diff
@@ -104,13 +104,19 @@ declare namespace BestOfJS {
+    username: string;
+  }
+
+  interface HallOfFameMember extends RawHallOfFameMember {
```
and this is the App's data model?

michaelrambeau replied:
And yes that is the populated object, the Hall of Game member with the related projects, ready to be consumed by the frontend application.

zawarudo reviewed (approved):
Everything looks good, just some clarification questions.

michaelrambeau commented:
## Second iteration

- Added: project logo (for projects that have a SVG logo, the "Featured Projects")
- Removed: number of followers
- Link to GitHub profile set on the GitHub username, in orange

![image](https://github.com/bestofjs/bestofjs/assets/5546996/f6338f00-f623-4da0-927c-4fc96bf9354b)
